### PR TITLE
ENG-26177 : Send the logmsgs for CWE collector

### PR DESCRIPTION
**Pull request by rakhimundhada15**
1. Set the stack name, application id for existing collector 
2. Override the register method to set the collector id for existing collector 
3. Override the process method to send the secmsgs,logmsgs and lmcstats for cwe collector.
4. Used al-aws-collector handleEvent  method and removed common code.

### CODE REVIEW

File: al-cwe-collector.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        let cweRegisterProps = this.getProperties(event);
+        AlAwsCollector.prototype.register.call(collector, event, cweRegisterProps, callback);
+    }
+
+    process(event, callback) {
+        const context = this._invokeContext;
+        var collector = this;
+        async.waterfall([
+            function (asyncCallback) {
+                collector._formatFun(event, context, asyncCallback);
```
Comment (kkuzmin) [inline al-cwe-collector.js:51]: let's modify [`formatFun`](https://github.com/alertlogic/cwe-collector/pull/66/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R85) to return a json object.

File: al-cwe-collector.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            },
+            function (formattedData, compress, asyncCallback) {
+                if (arguments.length === 2 && typeof compress === 'function') {
+                    asyncCallback = compress;
+                    compress = true;
+                }
+                collector.send(formattedData, compress, collector._ingestType, (err, res) => {
```
Comment (kkuzmin) [inline al-cwe-collector.js:58]: Then let's stringify `formattedData ` here

File: al-cwe-collector.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    return asyncCallback(err, formattedData);
+                });
+            },
+            function (formattedData, asyncCallback) {
+                let formatedJsonData = JSON.parse(formattedData);
```
Comment (kkuzmin) [inline al-cwe-collector.js:63]: and remove `parse()` here
in order to avoid double json conversion

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: al-cwe-collector.js (hunk 1 above)
Comment (rakhimundhada15) [reply al-cwe-collector.js:51]: Removed JSON stringify from format function and return original  json object.

**Review by kkuzmin — APPROVED** [review]

**Comment by kkuzmin** [comment]
test it

File: ps_spec.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            - pull_request
+            - pull_request:
+                trigger_phrase: test it
+        image: node:12
+        compute_size: small
+        commands:
+            - make test
+
+    -
+        name: Master Push - Publish
```
Comment (kkuzmin) [inline ps_spec.yml:17]: we don't need this step because we don't publish collector packages in npm

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: ps_spec.yml (hunk 4 above)
Comment (rakhimundhada15) [reply ps_spec.yml:17]: okay. I will update.

**Review by kkuzmin — APPROVED** [review]

**Pull request merged**